### PR TITLE
Added advance-message-interval option and more federated tests for TypeScript

### PR DESCRIPTION
**Pull request by hokeun**
Relevant reactor-ts PR: https://github.com/lf-lang/reactor-ts/pull/109

### CODE REVIEW

**Comment by hokeun** [comment]
Note: I think this change shouldn't cause any merge conflict on the fed-gen PR: https://github.com/lf-lang/lingua-franca/pull/1221

File: org.lflang/src/org/lflang/generator/ts/TSConstructorGenerator.kt [hunk 1]
```diff
@@ -105,6 +107,17 @@ class TSConstructorGenerator (
...
     }
 
+    // Generate code for setting target configurations.
+    private fun generateTargetConfigurations(): String {
+        val targetConfigurations = LinkedList<String>()
+        if ((reactor.isMain || reactor.isFederated) &&
+            targetConfig.coordinationOptions.advance_message_interval != null) {
+            targetConfigurations.add(
+                "this.setAdvanceMessageInterval(${timeInTargetLanguage(targetConfig.coordinationOptions.advance_message_interval)})")
+        }
```
Comment (Soroosh129) [inline org.lflang/src/org/lflang/generator/ts/TSConstructorGenerator.kt:117]: I think there should be a warning here if advance_message_interval is not set by the user. At least this is how this scenario is handled in C and Python.

**Review by Soroosh129 — APPROVED** [review]
LGTM!

File: org.lflang/src/org/lflang/generator/ts/TSConstructorGenerator.kt (hunk 1 above)
Comment (hokeun) [reply org.lflang/src/org/lflang/generator/ts/TSConstructorGenerator.kt:117]: Thank you for the suggestion! I added a warning here: https://github.com/lf-lang/lingua-franca/pull/1293/commits/1c6a598202f12cbf1b3ab902f5ad872a4d39f3af

**Pull request merged**